test/widgets/lightbox_test.dart

```diff
@@ -127,20 +228,24 @@ void main() {
             routeEntranceAnimation: kAlwaysCompleteAnimation,
             message: eg.streamMessage(),
             src: videoSrc)))));
-      await tester.pumpAndSettle();
+      await tester.pump(); // global store
+      await tester.pump(); // per-account store
+      await tester.pump(); // video controller initialization
     }
 
     testWidgets('shows a VideoPlayer, and video is playing', (tester) async {
-      await setupPage(tester, videoSrc: Uri.parse('https://a/b.mp4'));
+      await setupPage(tester, videoSrc: Uri.parse(FakeVideoPlayerPlatform.kTestVideoUrl));
 
       check(FakeVideoPlayerPlatform.initialized).isTrue();
       check(FakeVideoPlayerPlatform.isPlaying).isTrue();
 
       await tester.ensureVisible(find.byType(VideoPlayer));
+
+      FakeVideoPlayerPlatform.cancelTimer();
```

Comment:
Yeah generally one would think that tearDown callback should run before the timer leak checks, but unfortunately that isn't the case, which is understandable - see https://github.com/flutter/flutter/issues/24166

So, in this test case, the timer is created when the page is initialized and generally should cancel out when page is disposed, but since we don't have that setup (for simplicity of the test) the `dispose` isn't called on the VideoPlayerController, leading to a "timer leak".

Reply:
Anyway, I think pushing the lightbox page in a route and pop-ing it before the end of each test (just like how `cancelTimer` is called currently), would avoid this "hacky" looking `cancelTimer` function, so I'll change the tests behavior to that.

Reply:
(Also would like to get to know a better way of doing this, if I am missing something here)

Reply:
So, turns out dispose is being called, and my tentative workaround (of temporary page route) was completely unnecessary, good to know.

Also after debugging the tests, the timer leak checking actually is triggered in the dispose call, it happens [exactly here](https://github.com/flutter/packages/blob/7b423f58218025f770efac80485802c31039e4a6/packages/video_player/video_player/lib/video_player.dart#L513) which is surprising and seems like a magic behaviour (since it's implementation is internal to dart vm).

Anyway after leading to nowhere, I replaced the timer implementation to instead use stopwatch. And everything seems to be correct now.